docs/deployment-guides/parallel-replicas.mdx

```diff
@@ -0,0 +1,499 @@
+        Each shard reads, filters, and aggregates the data locally and then
+        sends back a mergeable state to the coordinator.
+    </li>
+    <li>
+        The coordinating node merges the data and then sends the response back
+        to the client.
+    </li>
+</ol>
+
+When we add replicas into the mix, the process is fairly similar, with the only
+difference being that only a single replica from each shard will execute the query.
+This means that more queries can then be processed in parallel.
+
+## Non-sharded architecture
+
+ClickHouse Cloud has a very different architecture to the one presented above.
+(See ["ClickHouse Cloud Architecture"](https://clickhouse.com/docs/cloud/reference/architecture)
+for more details). With separation of compute and storage, and with virtually an
+infinite amount of storage, the need for shards becomes less important.
+
+The figure below shows the ClickHouse Cloud architecture:
+
+![non-sharded architecture](@site/docs/deployment-guides/images/parallel-replicas-2.png)
+
+This architecture allows us to be able to add and remove replicas nearly
+instantaneously, ensuring a very high cluster scalability. The ClickHouse
+Keeper cluster (shown right) ensures that we have a single source of truth
+for the metadata. Replicas can fetch the metadata from the ClickHouse Keeper
+cluster and all maintain the same data. The data themselves are stored in
+object storage, and the SSD cache allows us to speed up queries.
+
+But how can we now distribute query execution across multiple servers? In a
+sharded architecture, it was fairly obvious given that each shard could actually
+execute a query on a subset of the data. How does it work when there is no sharding?
+
+## Introducing parallel replicas
+
+To parallelize query execution through multiple servers, we first need to be
+able to assign one of our servers as a coordinator. The coordinator is the one
+that creates the list of tasks that need to be executed, ensures they are all
+executed, aggregated and that the result is returned to the client. Like
+in most distributed systems, this will be the role of the node that receives the
+initial query. We also need to define the unit of work. In a sharded architecture,
+the unit of work is the shard, a subset of the data. With parallel replicas we
+will use a small portion of the table, called [granules](/docs/guides/best-practices/sparse-primary-indexes#data-is-organized-into-granules-for-parallel-data-processing),
+as the unit of work.
+
+Now, let's see how it works in practice with the help of the figure below:
+
+![Parallel replicas](@site/docs/deployment-guides/images/parallel-replicas-3.png)
+
+With parallel replicas:
+
+<ol className="docs-ordered-list">
+    <li>
+        The query from the client is sent to one node after going through a load
+        balancer. This node becomes the coordinator for this query.
+    </li>
+    <li>
+        The node analyzes the index, and selects the right parts and
```

Comment:
minor issue, but super correct would be "the node analyzes the index of each part"